README.md

```diff
@@ -0,0 +1,142 @@
+is `pactum` package itself and some knowledge of python.
+
+```python
+import pactum
+
+class MyAPI(API):
+    name = 'My API'
+    versions = [...]
+```
+
+### Easy to extend
+
+Using the [visitor pattern](http://wiki.c2.com/?VisitorPattern) you can create
+exporters and extensions for any format or service you want.
+
+Take a look at [pactum/exporters/openapi.py](pactum/exporters/openapi.py).
+
+### Architecture
+
+Always keep this diagram in mind when defining your APIs.
+
+![Architecture Diagram](Diagram.svg)
+
+
+### Tutorial
+
+Create a file called specs.py and start defining your API.
+
+You can define a `Resource` object for your API.
+
+```python
+from pactum import Action, API, Resource, Response, Version
+from pactum import fields, verbs
+
+class MyResource(Resource):
+    fields = [
+        fields.IntegerFieldField(name='code', required=True),
+        fields.StringField(name='item')
+    ]
+resource = MyResource()
+
+error_resource = Resource(
+    name = 'ErrorResource'
+    fields = [fields.StringField(name='error', required=False)]
+)
+```
+You can define any element of your specification by calling it directly as in
+`error_resource` or by class definition as in `MyResource` and then calling it.
+
+
+List resources are definitions of lists of the same resource.
+```python
+list_resource = MyListResource(resource=resource)
+```
+
+You can define `Response` objects with `status`, `description`(optional)  a
+`header`(optional) and a `Resource`/`ListResource` object as `body` (optionally)...
+
+```python
+ok_response = Response(
+    status=200, description='Here is your resource list.', body=list_resource
+)
+
+error_response = Response(status=400, resource=error_resource, headers=[('Content-type': 'application-json')])
+```
+
+... and `Request` objects with `verb`, `description`, `header`(optional) and a `Resource`/`ListResource`
+object as `payload`.
+
+```python
+request = Request(verb=verbs.GET, payload=resource)
+```
+
+An `Action` groups your request and a list of responses for an specified action
+passed in description parameter.
+```python
+action = Action(
+    description='Returns a list of resources.',
+    request=request, responses=[error_response, ok_response]
+)
+```
+The Action object, as all other elements in Pactum, receive a description string
+that sets the `.__doc__` attribute and can be the docstring of the class
+if the object is defined by class definition.
+
+A route can have a list of actions in an HTTP path.
+```python
+class ResourceRoute(Route):
+    path = '/resource'
+    action = [action]
+
+route = ResourceRoute()
+```
+
+Your routes can be grouped in API versions.
+```python
+class V1(Version):
+    name = 'V1'
+    routes = [route]
+
+v1 = V1()
+```
+Then you can define your API. ;)
+```python
+class MyAPI(API):
+    name = 'My API'
+    versions = [v1]
+
+api = MyAPI()
+```
+Be happy and ready to export your specification to any format you want.
```

Comment:
`any` meaning openapi? xD

Reply:
For now it is openapi. But it's easy to define an exporter for another format. This is what I want to mean here.
I can add a change that makes it more explicit.